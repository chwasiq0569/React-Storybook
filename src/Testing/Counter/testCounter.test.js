import React, { Component } from "react";
import TestCounter from "./TestCounter";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
//toBe is strict comparison
describe("Counter Component is Rendering Seccessfully", () => {
  it("Matched the snapshot", () => {
    //takes snapshot of component if any changes occurs in UI this will yell
    const tree = renderer.create(<TestCounter />).toJSON();
    //if any positive change happens then press u to update snapshot
    expect(tree).toMatchSnapshot();
  });
  it("starts with a Counter of 0", () => {
    const wrapper = shallow(<TestCounter />);
    const countState = wrapper.state().count;
    expect(countState).toEqual(0);
  });
  it("Increment Btn should contain +++ Text", () => {
    const wrapper = shallow(<TestCounter />);
    const incrementBtn = wrapper.find(`[data-test='incrementBtn']`);
    expect(incrementBtn.text()).toEqual("+++");
  });
  it("p tag should contain count: 0 initially", () => {
    const wrapper = shallow(<TestCounter />);
    const paragraphTag = wrapper.find(".countValue");
    expect(paragraphTag.text()).toEqual("count: 0");
  });
  it("should render text inside p tag to count: 2 on click", () => {
    const wrapper = shallow(<TestCounter />);
    const incrementBtn = wrapper.find(`[data-test='incrementBtn']`);
    incrementBtn.simulate("click");
    const paragraphTag = wrapper.find(".countValue");
    expect(paragraphTag.text()).toBe("count: 2");
  });
  it("should decrement count by 2 on decrementBtn click", () => {
    const wrapper = shallow(<TestCounter />);
    const decrementBtn = wrapper.find(`[data-test='decrementBtn']`);
    decrementBtn.simulate("click");
    expect(wrapper.state().count).toEqual(-2);
  });
});
