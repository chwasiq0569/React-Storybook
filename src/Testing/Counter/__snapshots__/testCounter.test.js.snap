// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component is Rendering Seccessfully Matched the snapshot 1`] = `
Array [
  <button
    data-test="incrementBtn"
    onClick={[Function]}
  >
    Increment
  </button>,
  <p
    className="countValue"
  >
    count: 
    0
  </p>,
  <button
    data-test="decrementBtn"
    onClick={[Function]}
  >
    Decrement
  </button>,
]
`;
